dice_ml/data_interfaces/public_data_interface.py

```diff
@@ -465,17 +464,25 @@ def prepare_query_instance(self, query_instance):
 
         elif isinstance(query_instance, pd.DataFrame):
             test = query_instance.copy()
-
         else:
             raise ValueError("Query instance should be a dict, a pandas dataframe, a list, or a list of dicts")
+        return test
 
+    def prepare_query_instance(self, query_instance):
+        """Prepares user defined test input(s) for DiCE."""
+        test = self.query_instance_to_df(query_instance)
         test = test.reset_index(drop=True)
         # encode categorical and numerical columns
         test = self._set_feature_dtypes(test,
                                         self.categorical_feature_names,
                                         self.continuous_feature_names)
         return test
 
+    def ensure_consistent_type(self, output_df, query_instance):
```

Comment:
Could you defined this function in https://github.com/interpretml/DiCE/blob/main/dice_ml/data_interfaces/base_data_interface.py? This will be inherited in both public and private data.

Reply:
Done